service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceMethodClientOptions.java

```diff
@@ -0,0 +1,39 @@
+package com.microsoft.azure.sdk.iot.service.devicetwin;
+
+import com.microsoft.azure.sdk.iot.service.ProxyOptions;
+import lombok.Builder;
+import lombok.Getter;
+
+/**
+ * Configurable options for all device method operations.
+ */
+@Builder
+public class DeviceMethodClientOptions
+{
+    protected static final Integer DEFAULT_HTTP_READ_TIMEOUT_MS = 24000; // 24 seconds
+    protected static final Integer DEFAULT_HTTP_CONNECT_TIMEOUT_MS = 24000; // 24 seconds
+
+    /**
+     * The options that specify what proxy to tunnel through. If null, no proxy will be used.
+     */
+    @Getter
+    private ProxyOptions proxyOptions;
+
+    /**
+     * The http read timeout to a specified timeout, in milliseconds. A non-zero value specifies the timeout when reading from
+     * Input stream after a connection is established to a resource. If the timeout expires before there is data available
+     * for read, a java.net.SocketTimeoutException is raised. A timeout of zero is interpreted as an infinite timeout.
```

Comment:
What happens if these are negative values?

Reply:
The http library will throw an exception saying "timeout cannot be negative"